front/src/components/oauth2/OAuth2Config.jsx

```diff
@@ -0,0 +1,119 @@
+                        </h2>
+                        {props.oauth2ErrorMsg && (
+                          <div class="alert alert-danger">
+                            <Text id={`integration.oauth2.${props.oauth2ErrorMsg}`} />
+                          </div>
+                        )}
+                        <p>
+                          <MarkupText id={`integration.${integrationName}.settings.introduction`} />
+                        </p>
+                        <p>
+                          <MarkupText id={`integration.oauth2.instructions`} />
+                        </p>
+                        <div class="form-group">
+                          <div class="form-label">
+                            <Text id={`integration.oauth2.apiKeyLabel`} />
+                          </div>
+                          <div class="input-group">
+                            <Localizer>
+                              <input
+                                type="text"
+                                class="form-control"
+                                placeholder={<Text id={`integration.oauth2.clientId`} />}
+                                onInput={props.updateClientId}
+                                value={props.clientId}
+                              />
+                            </Localizer>
+                          </div>
+                        </div>
+                        <div class="form-group">
+                          <div class="input-group">
+                            <Localizer>
+                              <input
+                                type="text"
+                                class="form-control"
+                                placeholder={<Text id={`integration.oauth2.secretId`} />}
+                                onInput={props.updateSecretId}
+                                value={props.secretId}
+                              />
+                            </Localizer>
+                          </div>
+                        </div>
+                        <div class="form-group">
+                          <span class="input-group-append">
+                            <button class="btn btn-primary" onClick={props.startConnect}>
+                              <Text id={`integration.oauth2.buttonConnect`} />
+                            </button>
+                          </span>
+                        </div>
+                      </div>
+                    )}
+                    {props.oauth2GetStatus === RequestStatus.Success && props.clientIdInDb && (
+                      <div class="dimmer-content">
+                        <h2>
+                          <Text id={`integration.${integrationName}.settings.title`} />
+                        </h2>
+                        <div class="alert alert-info">
+                          <p>
+                            <Text id={`integration.${integrationName}.settings.complete`} />
```

Comment:
I'm not sure this key is set for all integrations.
Isn't it a OAuth client key?

Reply:
> Some remarks on code and inconsistencies.
> 
> To get a shortcut, this feature should:
> 
>     1. store 3 vars in db for a service (client_id, client_secret, auth_code)
> 
>     2. generates an access_token uri
> 
>     3. generates an authorization uri
> 
> 
> I'm a right?

yes you are right,
+ the capability to execute http request with the oauth2 token (and refresh token if its expired)

the objective of this pr is to write generic api to request third party platform with aouth2 authentication and a simple form (with client id and secret)  for integration (optional, is possible to write specific formula if the developer want to do), it's why you can't find  code for   some call (like        const returnSaveVar = await state.httpClient.post(`/api/v1/service/${state.integrationName}/savevar`, { )

I try to answer/change all point later